File: 1-TwoSum/1-TwoSum.java
// Last updated: 16/10/2025, 04:32:00
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> mp = new HashMap();
        int n = nums.length;
        int ans[] = new int[2];
        for(int i=0;i<n;i++){
            int b = target - nums[i];
            if(mp.containsKey(b)){
                ans[0] = i;
                ans[1] = mp.get(b);
                break;
            }else{
                mp.put(nums[i],i);
            }
        }
        return ans;
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 16/10/2025, 04:31:59
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int> mp;
        int sum = 0;
        for(int i=0;i<n;i++){
            int b = target - nums[i];
            if(mp.find(b) != mp.end()){
                return {i,mp[b]};
            }
            mp[nums[i]] = i;
            
        }
        return {-1,-1};
    }
};
    
    
            

        